extended-tracer/src/main/java/io/opentelemetry/contrib/tracer/Tracing.java

```diff
@@ -0,0 +1,374 @@
+   *
+   * <p>If an exception is thrown by the {@link SpanCallback}, the span will be marked as error, and
+   * the exception will be recorded.
+   *
+   * @param spanBuilder the {@link SpanBuilder} to use
+   * @param spanCallback the {@link SpanCallback} to call
+   * @param <T> the type of the result
+   * @param <E> the type of the exception
+   * @return the result of the {@link SpanCallback}
+   */
+  public <T, E extends Throwable> T call(SpanBuilder spanBuilder, SpanCallback<T, E> spanCallback)
+      throws E {
+    return call(spanBuilder, spanCallback, Tracing::setSpanError);
+  }
+
+  /**
+   * Runs the given {@link SpanCallback} inside of the span created by the given {@link
+   * SpanBuilder}. The span will be ended at the end of the {@link SpanCallback}.
+   *
+   * <p>If an exception is thrown by the {@link SpanCallback}, the handleException consumer will be
+   * called, giving you the opportunity to handle the exception and span in a custom way, e.g. not
+   * marking the span as error.
+   *
+   * @param spanBuilder the {@link SpanBuilder} to use
+   * @param spanCallback the {@link SpanCallback} to call
+   * @param handleException the consumer to call when an exception is thrown
+   * @param <T> the type of the result
+   * @param <E> the type of the exception
+   * @return the result of the {@link SpanCallback}
+   */
+  @SuppressWarnings("NullAway")
+  public <T, E extends Throwable> T call(
+      SpanBuilder spanBuilder,
+      SpanCallback<T, E> spanCallback,
+      BiConsumer<Span, Throwable> handleException)
+      throws E {
+    Span span = spanBuilder.startSpan();
+    //noinspection unused
+    try (Scope unused = span.makeCurrent()) {
+      return spanCallback.doInSpan();
+    } catch (Throwable e) {
+      handleException.accept(span, e);
+      throw e;
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * Marks a span as error.
+   *
+   * @param span the span
+   * @param exception the exception that caused the error
+   */
+  private static void setSpanError(Span span, Throwable exception) {
+    span.setStatus(StatusCode.ERROR);
+    span.recordException(exception);
+  }
+
+  /**
+   * Injects the current context into a string map, which can then be added to HTTP headers or the
+   * metadata of an event.
+   */
+  public Map<String, String> getPropagationHeaders() {
+    Map<String, String> transport = new HashMap<>();
+    //noinspection ConstantConditions
+    openTelemetry
+        .getPropagators()
+        .getTextMapPropagator()
+        .inject(
+            Context.current(),
+            transport,
+            (map, key, value) -> {
+              if (map != null) {
+                map.put(key, value);
+              }
+            });
+
+    return transport;
+  }
+
+  /**
+   * Extract the context from a string map, which you get from HTTP headers of the metadata of an
+   * event you're processing.
+   */
+  private Context extractContext(Map<String, String> transport) {
```

Comment:
If you go with `getTextMapPropagationContext` as suggested above, maybe this should be `extractTextMapPropagationContext()`?

Its weird that in order to access this method, which is purely related to context propagation, you need to configure an object called `Tracing` and include information to configure a `Tracer`.

What do you think about creating a second object called `Propagation` to hold the methods that are purely propagation related?

Reply:
it's really just that one method - now called `getTextMapPropagationContext`.

I'm not sure it helps discoverability if we move this method to a different class.

Reply:
> I'm not sure it helps discoverability if we move this method to a different class.

Yeah it definitely hurts. What about making the methods which don't depend on `Tracer` be pure functions? I.e. public static method which accept `OpenTelemetry` as a parameter. That way, users who wish to benefit from their convenience but don't want to use any of the methods that depend on `Tracer` can use them without having to create a dummy tracer.

Reply:
I can make this one method a static method in `Propagation` (as you suggested) - and leave the existing method as-is.

Reply:
`callWithBaggage` can be made static as well.

Reply:
done @jack-berg